src/runmode-dpdk.c

```diff
@@ -0,0 +1,1295 @@
+    }
+    retval = ConfigSetCopyIfaceSettings(iconf, copy_iface_str, copy_mode_str);
+    if (retval < 0) {
+        return retval;
+    }
+
+    if (iconf->copy_mode == DPDK_COPY_MODE_NONE) {
+        SCLogInfo("IDS mode enabled, disabling TX settings");
+        iconf->nb_tx_queues = 0;
+        iconf->nb_tx_desc = 0;
+    }
+
+    SCReturnInt(0);
+}
+DPDKIfaceConfig *ConfigParse(const char *iface)
+{
+    int retval;
+    DPDKIfaceConfig *iconf = NULL;
+    if (iface == NULL) {
+        SCLogError(SC_ERR_DPDK_CONF, "Provided iface is NULL");
+        SCReturnPtr(NULL, "void *");
+    }
+
+    ConfigInit(&iconf);
+    if (iconf == NULL) {
+        SCReturnPtr(NULL, "void *");
+    }
+    retval = ConfigLoad(iconf, iface);
+    if (retval < 0) {
+        iconf->DerefFunc(iconf);
+        SCReturnPtr(NULL, "void *");
+    }
+
+    return iconf;
+}
+
+void DeviceInitPortConf(const DPDKIfaceConfig *iconf, const struct rte_eth_dev_info *dev_info,
+        struct rte_eth_conf *port_conf)
+{
+    *port_conf = (struct rte_eth_conf){
+            .rxmode = {
+                    .mq_mode = ETH_MQ_RX_NONE,
+                    .max_rx_pkt_len = RTE_ETHER_MAX_LEN,
+                    .offloads = 0, // turn every offload off to prevent any packet modification
+            },
+            .txmode = {
+                    .mq_mode = ETH_MQ_TX_NONE,
+                    .offloads = 0,
+            },
+    };
+
+    // configure RX offloads
+    if (dev_info->rx_offload_capa & DEV_RX_OFFLOAD_RSS_HASH) {
+        if (iconf->nb_rx_queues > 1) {
+            SCLogConfig("RSS enabled on %s for %d queues", iconf->iface, iconf->nb_rx_queues);
+            port_conf->rx_adv_conf.rss_conf.rss_key = rss_hkey;
+            port_conf->rx_adv_conf.rss_conf.rss_key_len = RSS_HKEY_LEN;
+            port_conf->rx_adv_conf.rss_conf.rss_hf = ETH_RSS_TCP | ETH_RSS_UDP;
+            port_conf->rxmode.mq_mode = ETH_MQ_RX_RSS;
+        } else {
+            SCLogConfig("RSS not enabled on %s", iconf->iface);
+            port_conf->rx_adv_conf.rss_conf.rss_key = NULL;
+            port_conf->rx_adv_conf.rss_conf.rss_hf = 0;
+        }
+    } else {
+        SCLogConfig("RSS not supported on %s", iconf->iface);
+    }
+
+    if (iconf->checksum_mode == CHECKSUM_VALIDATION_DISABLE) {
+        SCLogConfig("Checksum validation disabled on %s", iconf->iface);
+    } else if (dev_info->rx_offload_capa & DEV_RX_OFFLOAD_CHECKSUM) {
+        if (iconf->checksum_mode == CHECKSUM_VALIDATION_ENABLE &&
+                iconf->flags & DPDK_RX_CHECKSUM_OFFLOAD) {
+            SCLogConfig("IP, TCP and UDP checksum validation enabled and offloaded "
+                        "on %s",
+                    iconf->iface);
+            port_conf->rxmode.offloads |= DEV_RX_OFFLOAD_CHECKSUM;
+        } else if (iconf->checksum_mode == CHECKSUM_VALIDATION_ENABLE &&
+                   !(iconf->flags & DPDK_RX_CHECKSUM_OFFLOAD)) {
+            SCLogConfig("Suricata checksum validation enabled "
```

Comment:
not sure if this warning or error. if it s a warning I am unable to find rx_callabck registration to check checksum offload to Software. please rework

Reply:
To support vendor-neutral code there is a check on line 872. If the NIC does not support checksum validation then it is validated by Suricata. 
The line 881 should only inform the user that while checksum validation is enabled it can also be offloaded to the NIC. 
Reworded to:
`SCLogConfig("Suricata checksum validation enabled (but can be offloaded on %s)", iconf->iface);`